azure_img_utils/cli/blob.py

```diff
@@ -0,0 +1,289 @@
+    '--page-blob',
+    is_flag=True,
+    default=False,
+    help='The image to upload is of page blob type. '
+)
+@click.option(
+    '--expand-image',
+    is_flag=True,
+    default=False,
+    help='The image to upload should be expanded. '
+)
+@click.option(
+    '--max-workers',
+    type=click.IntRange(min=1),
+    default=None,
+    help='Maximum number of workers allowed for upload. '
+)
+@click.option(
+    '--max-retry-attempts',
+    type=click.IntRange(min=0),
+    default=None,
+    help='Maximum retry attempts for upload. '
+)
+@add_options(shared_options)
+@click.pass_context
+def upload(
+    context,
+    storage_account,
+    blob_name,
+    container,
+    image_file,
+    force_replace_image,
+    page_blob,
+    expand_image,
+    max_workers,
+    max_retry_attempts,
+    **kwargs
+):
+    """
+    Uploads an file as blob to the specified container
+    """
+    process_shared_options(context.obj, kwargs)
+    config_data = get_config(context.obj)
+    logger = logging.getLogger('azure_img_utils')
+    logger.setLevel(config_data.log_level)
+    try:
+        az_img = AzureImage(
+            container=container,
+            storage_account=storage_account,
+            credentials_file=config_data.credentials_file,
+            resource_group=config_data.resource_group,
+            log_level=config_data.log_level,
+            log_callback=logger
+        )
+        blob_name = az_img.upload_image_blob(
+            image_file,
+            max_workers=max_workers,
+            max_retry_attempts=max_retry_attempts,
+            blob_name=blob_name,
+            force_replace_image=force_replace_image,
+            page_blob=page_blob,
+            expand_image=expand_image
+        )
+        if blob_name and config_data.log_level != logging.ERROR:
+            echo_style(
+                f'blob {blob_name} uploaded',
+                config_data.no_color,
+                fg='green'
+            )
+        else:
+            echo_style(
+                f'unable to upload blob {blob_name}',
+                config_data.no_color
+            )
+
+    except Exception as e:
+        echo_style(
+            'Unable to upload blob',
+            config_data.no_color,
+            fg='red'
+        )
+        echo_style(str(e), config_data.no_color, fg='red')
+        sys.exit(1)
+
+
+# -----------------------------------------------------------------------------
+# Blob delete commands function
+@blob.command()
+@click.option(
+    '--storage-account',
+    type=click.STRING,
+    required=True,
+    help='Storage account for the blobs.'
+)
+@click.option(
+    '--blob-name',
+    type=click.STRING,
+    required=True,
+    help='Name of the blob to check.'
+)
+@click.option(
+    '--container',
+    type=click.STRING,
+    required=True,
+    help='Container for the blob to check.'
+)
+@add_options(shared_options)
+@click.confirmation_option(
+    help='This command will delete the specified blob. Are you sure?')
```

Comment:
Nitpick: Please carry the closing bracket to a new line.

Reply:
Done!